server/cmd_auth.go

```diff
@@ -18,15 +18,30 @@ type Select struct {
 	commands.Select
 }
 
-func (cmd *Select) Handle(conn Conn) error {
+func (cmd *Select) Handle(conn Conn) (err error) {
 	ctx := conn.Context()
+
+	// As per RFC1730#6.3.1,
+	// 		The SELECT command automatically deselects any
+	// 		currently selected mailbox before attempting the new selection.
+	// 		Consequently, if a mailbox is selected and a SELECT command that
+	// 		fails is attempted, no mailbox is selected.
+	// Thus if SELECT fails for whatever reason we should unselect.
+	// For example, some clients (e.g. Apple Mail) perform SELECT "" when the
+	// server doesn't announce the UNSELECT capability.
+	defer func() {
```

Comment:
This doesn't build.

I think it's fine to just update the context's current mailbox at the top of the function.

Reply:
The race you're referring to is a separate bug. Apparently we didn't have an issue for it, filled one: https://github.com/emersion/go-imap/issues/367

Reply:
Done. Now all tests pass.